gradle quarkusBuild  has error : SRCFG00035: Failed to load resource

Since upgrading from to 1.13.x, run below command
```
./gradlew  quarkusBuild -Dquarkus.package.type=uber-jar  -Dorg.gradle.jvmargs=-Xmx256m
```
Gradle made a error: SRCFG00035: Failed to load resource, full stacktrace is:
```
> Task :processResources

> Task :classes
start copy some files...

> Task :inspectClassesForKotlinIC
> Task :jar

> Task :quarkusBuild
building quarkus jar

> Task :quarkusBuild FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':quarkusBuild'.
> java.lang.IllegalStateException: SRCFG00035: Failed to load resource

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
```
My system jdk version :
```
openjdk 11.0.10 2021-01-19
OpenJDK Runtime Environment 18.9 (build 11.0.10+9)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.10+9, mixed mode, sharing)
```
when I use `adoptjdk15-openj9` or downgrade quarkus version to `1.12.2.Final`, it' all fine.

I have same error code.

`gradle clean build --debug --stacktrace`
```
2021-04-16T07:32:05.732+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at io.smallrye.config.AbstractLocationConfigSourceLoader.tryFileSystem(AbstractLocationConfigSourceLoader.java:100)
2021-04-16T07:32:05.742+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at io.smallrye.config.AbstractLocationConfigSourceLoader.loadConfigSources(AbstractLocationConfigSourceLoader.java:81)
2021-04-16T07:32:05.742+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at io.smallrye.config.AbstractLocationConfigSourceLoader.loadConfigSources(AbstractLocationConfigSourceLoader.java:65)
2021-04-16T07:32:05.742+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at io.smallrye.config.DotEnvConfigSourceProvider.getConfigSources(DotEnvConfigSourceProvider.java:45)
2021-04-16T07:32:05.742+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at io.quarkus.runtime.configuration.ConfigUtils.configBuilder(ConfigUtils.java:108)
2021-04-16T07:32:05.742+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at io.quarkus.runtime.configuration.ConfigUtils.configBuilder(ConfigUtils.java:67)
2021-04-16T07:32:05.742+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at io.quarkus.deployment.ExtensionLoader.loadStepsFrom(ExtensionLoader.java:146)
2021-04-16T07:32:05.742+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:101)
2021-04-16T07:32:05.742+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:394)
2021-04-16T07:32:05.743+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   ... 124 more
2021-04-16T07:32:05.743+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Caused by: java.nio.file.NoSuchFileException: /home/rpr/.gradle/daemon/7.0/.env
2021-04-16T07:32:05.743+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at io.smallrye.common.classloader.ClassPathUtils.readStream(ClassPathUtils.java:205)
2021-04-16T07:32:05.743+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at io.smallrye.common.classloader.ClassPathUtils.consumeStream(ClassPathUtils.java:158)
2021-04-16T07:32:05.743+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at io.smallrye.config.common.utils.ConfigSourceUtil.urlToMap(ConfigSourceUtil.java:64)
2021-04-16T07:32:05.743+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at io.quarkus.runtime.configuration.ConfigUtils$BuildTimeDotEnvConfigSourceProvider.loadConfigSource(ConfigUtils.java:202)
2021-04-16T07:32:05.743+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at io.smallrye.config.DotEnvConfigSourceProvider.loadConfigSource(DotEnvConfigSourceProvider.java:30)
2021-04-16T07:32:05.743+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at io.smallrye.config.AbstractLocationConfigSourceLoader.addConfigSource(AbstractLocationConfigSourceLoader.java:206)
2021-04-16T07:32:05.743+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   ... 140 more
2021-04-16T07:32:05.743+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
2021-04-16T07:32:05.743+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
2021-04-16T07:32:05.743+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Get more help at https://help.gradle.org
2021-04-16T07:32:05.743+0200 [ERROR] [org.gradle.internal.buildevents.BuildResultLogger] 
2021-04-16T07:32:05.743+0200 [ERROR] [org.gradle.internal.buildevents.BuildResultLogger] BUILD FAILED in 16s
```

I needed to add symlink from project to `/home/rpr/.gradle/daemon/7.0/.env`

Experiencing similar problems, but with `config/application.properties` instead of `.env`:

```
$ ./gradlew clean build
...
Caused by: java.nio.file.NoSuchFileException: /home/gerrit/.gradle/daemon/6.8.3/config/application.properties
...
```

Issue was introduced from `1.13.0.Final` until the actual version:

```shell
[x80486@uplink:~/somewhere/kotlin-quarkus]$ ./gradlew -Dquarkus.package.type=fast-jar --stacktrace clean quarkusBuild 
...
Caused by: java.lang.IllegalStateException: SRCFG00035: Failed to load resource
        at io.smallrye.config.AbstractLocationConfigSourceLoader.addConfigSource(AbstractLocationConfigSourceLoader.java:209)
        at io.smallrye.config.AbstractLocationConfigSourceLoader.addConfigSource(AbstractLocationConfigSourceLoader.java:201)
        at io.smallrye.config.AbstractLocationConfigSourceLoader.access$300(AbstractLocationConfigSourceLoader.java:49)
        at io.smallrye.config.AbstractLocationConfigSourceLoader$ConfigSourcePathConsumer.accept(AbstractLocationConfigSourceLoader.java:290)
        at io.smallrye.config.AbstractLocationConfigSourceLoader$ConfigSourcePathConsumer.accept(AbstractLocationConfigSourceLoader.java:279)
        at io.smallrye.common.classloader.ClassPathUtils.lambda$consumeAsPath$0(ClassPathUtils.java:102)
        at io.smallrye.common.classloader.ClassPathUtils.processAsPath(ClassPathUtils.java:140)
        at io.smallrye.common.classloader.ClassPathUtils.consumeAsPath(ClassPathUtils.java:101)
        at io.smallrye.config.AbstractLocationConfigSourceLoader.tryFileSystem(AbstractLocationConfigSourceLoader.java:100)
        at io.smallrye.config.AbstractLocationConfigSourceLoader.loadConfigSources(AbstractLocationConfigSourceLoader.java:81)
        at io.smallrye.config.AbstractLocationConfigSourceLoader.loadConfigSources(AbstractLocationConfigSourceLoader.java:65)
        at io.smallrye.config.DotEnvConfigSourceProvider.getConfigSources(DotEnvConfigSourceProvider.java:45)
        at io.quarkus.runtime.configuration.ConfigUtils.configBuilder(ConfigUtils.java:105)
        at io.quarkus.runtime.configuration.ConfigUtils.configBuilder(ConfigUtils.java:64)
        at io.quarkus.deployment.ExtensionLoader.loadStepsFrom(ExtensionLoader.java:146)
        at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:101)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:394)
        ... 124 more
Caused by: java.nio.file.NoSuchFileException: /home/x80486/.gradle/daemon/7.0/.env
        at io.smallrye.common.classloader.ClassPathUtils.readStream(ClassPathUtils.java:205)
        at io.smallrye.common.classloader.ClassPathUtils.consumeStream(ClassPathUtils.java:158)
        at io.smallrye.config.common.utils.ConfigSourceUtil.urlToMap(ConfigSourceUtil.java:64)
        at io.quarkus.runtime.configuration.ConfigUtils$BuildTimeDotEnvConfigSourceProvider.loadConfigSource(ConfigUtils.java:181)
        at io.smallrye.config.DotEnvConfigSourceProvider.loadConfigSource(DotEnvConfigSourceProvider.java:30)
        at io.smallrye.config.AbstractLocationConfigSourceLoader.addConfigSource(AbstractLocationConfigSourceLoader.java:206)
        ... 140 more
```

I tested all those versions one by one 😅 ...`1.12.2.Final` is fine.

--- a/core/runtime/src/main/java/io/quarkus/runtime/configuration/ApplicationPropertiesConfigSourceLoader.java
+++ b/core/runtime/src/main/java/io/quarkus/runtime/configuration/ApplicationPropertiesConfigSourceLoader.java
@@ -3,6 +3,7 @@ package io.quarkus.runtime.configuration;
 import java.io.IOException;
 import java.net.URI;
 import java.net.URL;
+import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.List;
 
@@ -48,7 +49,9 @@ public class ApplicationPropertiesConfigSourceLoader extends AbstractLocationCon
 
         @Override
         public List<ConfigSource> getConfigSources(final ClassLoader classLoader) {
-            return loadConfigSources("config/application.properties", classLoader);
+            return loadConfigSources(
+                    Paths.get(System.getProperty("user.dir"), "config", "application.properties").toUri().toString(),
+                    classLoader);
         }
 
         @Override

--- a/core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigUtils.java
+++ b/core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigUtils.java
@@ -1,7 +1,6 @@
 package io.quarkus.runtime.configuration;
 
 import static io.smallrye.config.AbstractLocationConfigSourceFactory.SMALLRYE_LOCATIONS;
-import static io.smallrye.config.DotEnvConfigSourceProvider.dotEnvSources;
 import static io.smallrye.config.ProfileConfigSourceInterceptor.SMALLRYE_PROFILE;
 import static io.smallrye.config.ProfileConfigSourceInterceptor.SMALLRYE_PROFILE_PARENT;
 import static io.smallrye.config.PropertiesConfigSourceProvider.classPathSources;
@@ -10,6 +9,7 @@ import static io.smallrye.config.SmallRyeConfigBuilder.WEB_INF_MICROPROFILE_CONF
 
 import java.io.IOException;
 import java.net.URL;
+import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.HashMap;
@@ -100,7 +100,9 @@ public final class ConfigUtils {
         builder.addDefaultInterceptors();
         if (runTime) {
             builder.addDefaultSources();
-            builder.withSources(dotEnvSources(classLoader));
+            builder.withSources(
+                    new DotEnvConfigSourceProvider(Paths.get(System.getProperty("user.dir"), ".env").toUri().toString())
+                            .getConfigSources(classLoader));
         } else {
             final List<ConfigSource> sources = new ArrayList<>();
             sources.addAll(classPathSources(META_INF_MICROPROFILE_CONFIG_PROPERTIES, classLoader));
@@ -190,7 +192,7 @@ public final class ConfigUtils {
      */
     static class BuildTimeDotEnvConfigSourceProvider extends DotEnvConfigSourceProvider {
         public BuildTimeDotEnvConfigSourceProvider() {
-            super();
+            super(Paths.get(System.getProperty("user.dir"), ".env").toUri().toString());
         }
 
         public BuildTimeDotEnvConfigSourceProvider(final String location) {

